### DISCUSSION TITLE
PrimaryKey in joined collections

### DISCUSSION THREAD

**Discussion by GrommasMario** [mikro-orm/mikro-orm#2713] [Q&A] [answered]
After serialization I get the primary key value in joined field (without populate), is it possible to change on the primary field?

In now
```json
{
  "id": 10,
  "title": "title",
  "description": "description",
  "category": 5
}
```
expected
```json
{
  "id": 10,
  "title": "title",
  "description": "description",
  "category": {
      "id":5
    }
}
```

I found code in v4.5.10 core/src/EntityComparator.ts, method: getPkGetter
```javascript
    } else {
      const pk = meta.primaryKeys[0];

      if (meta.properties[pk].reference !== ReferenceType.SCALAR) {
        lines.push(`  if (entity.${pk} != null && (entity.${pk}.__entity || entity.${pk}.__reference)) return entity.${pk}.__helper.getPrimaryKey();`);
      }

      lines.push(`  return entity.${pk};`);
    }
```
I got the desired result after replacing
```javascript
      lines.push(`  return { ${pk}: entity.${pk} }`);
    }
```

Is it possible to change to my version?

**Comment by B4nan** [comment, marked as answer]
You can use property serializer, or custom toJSON method.

https://mikro-orm.io/docs/serializing/#property-serializers

Changing internals like this is definitely not an option. We could have a config toggle for this, but definitely not change the default, that would be quite huge breaking change.

**Reply by GrommasMario** [reply]
Config toggle is good solution
